Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1129,3 +1112,10 @@ extension sourcekitd_api_uid_t {
     }
   }
 }
+
+extension SourceEdit {
+  var positiveLength: Int {
+    // Required since SourceEdit returns a negative length
+    -length.utf8Length
+  }
```

Comment:
Oh, we should fix this in SwiftSyntax. Could you open a PR in swift-syntax to change that?

Reply:
Yeah of course. However the comment there was a tiny bit misleading which made me think that it should be zero if the value is positive

```swift
  /// Length of the original source range that this edit applies to. Zero if
  /// this is an addition.
  public var length: SourceLength {
    return SourceLength(utf8Length: range.lowerBound.utf8Offset - range.upperBound.utf8Offset)
  }
 ```

Reply:
The `Zero if this is an addition` meant that we represent an assertion at offset 5 by a replacement of range `5..<5` by the new text.